client/src/app/components/TargetCard.tsx

```diff
@@ -105,23 +108,23 @@ export const TargetCard: React.FC<TargetCardProps> = ({
     }
   };
 
-  const getImage = (): React.ComponentType => {
-    let result: React.ComponentType<any> = CubesIcon;
-    const imagePath = target?.image?.id
-      ? `/hub/files/${target?.image.id}`
-      : DefaultRulesetIcon;
-    if (target.image) {
-      result = () => (
-        <img
-          src={imagePath}
-          alt="Card logo"
-          style={{ height: 80, pointerEvents: "none" }}
-        />
-      );
+  const fetchImageAndSetDataUrl = async (imagePath: string) => {
+    try {
+      const imageDataResponse = await axios.get(imagePath, {
+        headers: {
+          Accept: "application/octet-stream",
+        },
+      });
+      const encodedSvg = encodeURIComponent(imageDataResponse.data);
+      setImageDataUrl(`data:image/svg+xml,${encodedSvg}`);
+    } catch (error) {
+      console.error("There was an issue fetching the image:", error);
     }
-
-    return result;
   };
+  const imagePath = target?.image?.id
+    ? `/hub/files/${target?.image.id}`
+    : DefaultRulesetIcon;
+  fetchImageAndSetDataUrl(imagePath);
```

Comment:
So I think this can be swapped out for a `useEffect()` so it doesn't run too often...

```jsx
  useEffect(() => {
    const imagePath = target?.image?.id
      ? `/hub/files/${target?.image.id}`
      : DefaultRulesetIcon;

    // sneaky self executing async function since useEffect() needs a sync function
    (async () => {
      try {
        const imageDataResponse = await axios.get(imagePath, {
          headers: {
            Accept: "application/octet-stream",
          },
        });
        const encodedSvg = encodeURIComponent(imageDataResponse.data);
        setImageDataUrl(`data:${imageDataResponse.headers["content-type"]},${encodedSvg}`);
      } catch (error) {
        console.error("There was an issue fetching the image:", error);
      }
    })();
  }, [target, setImageDataUrl]);
```